process_report/config.py

```diff
@@ -0,0 +1,184 @@
+            self.OLD_PI_FILEPATH = util.fetch_s3(self.PI_S3_FILEPATH)
+        return self.OLD_PI_FILEPATH
+
+    def get_alias_filepath(self) -> pydantic.FilePath:
+        if not self.ALIAS_FILEPATH:
+            self.ALIAS_FILEPATH = util.fetch_s3(self.ALIAS_S3_FILEPATH)
+        return self.ALIAS_FILEPATH
+
+    def get_new_pi_credit_amount(self) -> Decimal:
+        if not self.NEW_PI_CREDIT_AMOUNT:
+            self.NEW_PI_CREDIT_AMOUNT = get_rates_info().get_value_at(
+                "New PI Credit", self.INVOICE_MONTH, Decimal
+            )
+        return self.NEW_PI_CREDIT_AMOUNT
+
+    def get_limit_new_pi_credit_to_partners(self) -> bool:
+        if not self.LIMIT_NEW_PI_CREDIT_TO_PARTNERS:
+            self.LIMIT_NEW_PI_CREDIT_TO_PARTNERS = get_rates_info().get_value_at(
+                "Limit New PI Credit to MGHPCC Partners", self.INVOICE_MONTH, bool
+            )
+        return self.LIMIT_NEW_PI_CREDIT_TO_PARTNERS
+
+    def get_bu_subsidy_amount(self) -> Decimal:
+        if not self.BU_SUBSIDY_AMOUNT:
+            self.BU_SUBSIDY_AMOUNT = get_rates_info().get_value_at(
+                "BU Subsidy", self.INVOICE_MONTH, Decimal
+            )
+        return self.BU_SUBSIDY_AMOUNT
+
+    def get_lenovo_su_charge_info(self) -> dict[str, Decimal]:
+        if not self.LENOVO_CHARGE_INFO:
+            self.LENOVO_CHARGE_INFO = {}
+            for su_name in ["GPUA100SXM4", "GPUH100"]:
+                self.LENOVO_CHARGE_INFO[su_name] = get_rates_info().get_value_at(
+                    f"Lenovo {su_name} Charge", self.INVOICE_MONTH, Decimal
+                )
+        return self.LENOVO_CHARGE_INFO
+
+    # Some invoices/processors want files loaded in certain ways
+    def get_nonbillable_pis(self) -> list[str]:
+        with open(self.NONBILLABLE_PIS_FILEPATH) as file:
+            return [line.rstrip() for line in file]
+
+    def get_nonbillable_projects(self) -> list[str]:
+        """Returns list of nonbillable projects for current invoice month"""
+        with open(self.NONBILLABLE_PROJECTS_FILEPATH) as file:
+            projects = [line.rstrip() for line in file]
+
+        timed_projects_list = self.get_nonbillable_timed_projects()
+        return list(set(projects + timed_projects_list))
+
+    def get_nonbillable_timed_projects(self) -> list[str]:
+        """Returns list of projects that should be excluded based on dates"""
+        dataframe = pandas.read_csv(self.NONBILLABLE_TIMED_PROECTS_FILEPATH)
+
+        # convert to pandas timestamp objects
+        dataframe["Start Date"] = pandas.to_datetime(
+            dataframe["Start Date"], format="%Y-%m"
+        )
+        dataframe["End Date"] = pandas.to_datetime(
+            dataframe["End Date"], format="%Y-%m"
+        )
+
+        mask = (dataframe["Start Date"] <= self.INVOICE_MONTH) & (
+            self.INVOICE_MONTH <= dataframe["End Date"]
+        )
+        return dataframe[mask]["Project"].to_list()
+
+    def get_alias_map(self) -> dict:
```

Comment:
Generally, we expect to see type arguments, like `dict[str, str]`, rather than an untyped `dict`.